apps/projects/app/script.js

```diff
@@ -296,15 +308,27 @@ function loadRequestsData({repoId, issueNumber}) {
 
 function getRequest(repoId, issueNumber, applicantId) {
   return new Promise(resolve => {
-    app.call('getApplicant', repoId, issueNumber, applicantId).subscribe( (address) => {
-      app.call('getAssignmentRequest', repoId, issueNumber, address).subscribe( (hash) => {
-        let contentJSON
-        ipfs.get(hash, (err, files) => {
-          for(const file of files) {
-            contentJSON = JSON.parse(file.content.toString('utf8'))
-          }
-          resolve({contributorAddr: address, ...contentJSON})
-        })
+    app.call('getApplicant', repoId, issueNumber, applicantId).subscribe( ({address, hash}) => {
+      let contentJSON
+      ipfs.get(hash, (err, files) => {
+        for(const file of files) {
+          contentJSON = JSON.parse(file.content.toString('utf8'))
+        }
+        resolve({contributorAddr: address, ...contentJSON})
+      })
+    })
+  })
+}
+
+function loadSubmissionData({issueNumber, repoId}, assignee) {
+  return new Promise(resolve => {
+    app.call('getSubmission', repoId, issueNumber, assignee).subscribe(async ({submissionHash, fulfillmentId, status}) => {
```

Comment:
the function passed as the subscription to getSubmission probably doesn't need to be async

Reply:
Yeah... try it without it and if it works then delete it?